packages/cli/src/lib/bundler/scalprumConfig.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DynamicRemotePlugin } from '@openshift/dynamic-plugin-sdk-webpack';
+import ESLintPlugin from 'eslint-webpack-plugin';
+import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
+import webpack, { ProvidePlugin } from 'webpack';
+
+import { resolve as resolvePath } from 'path';
+
+import { BundlingPaths } from './paths';
+import { transforms } from './transforms';
+import { DynamicPluginOptions } from './types';
+
+/**
+ * TODO: Create a config API to configure and optimize dependency sharing
+ * https://medium.com/@marvusm.mmi/webpack-module-federation-think-twice-before-sharing-a-dependency-18b3b0e352cb
+ */
+export const sharedModules = {
+  /**
+   * Mandatory singleton packages for sharing
+   */
+  react: {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  'react-dom': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  'react-router-dom': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@backstage/version-bridge': {
+    singleton: true,
+    requiredVersion: '*',
+  },
+  '@backstage/core-plugin-api': {
+    singleton: true,
+    requiredVersion: '*',
+  },
```

Comment:
Let's drop these, they're intended to not be singletons and this might actually break things